### DISCUSSION TITLE
GET_SALES_AND_TRAFFIC_REPORT VendorGroupID error

### DISCUSSION THREAD

**Discussion by THPheadnerd** [amzn/selling-partner-api-models#3390] [General] [answered]
Hello,

With the Amazon connection that I'm using we were able to create the GET_VENDOR_SALES_REPORT. Now, I'm trying to create the GET_SALES_AND_TRAFFIC_REPORT and am getting the following error:

FikaAmazonAPI.AmazonSpApiSDK.Models.Exceptions.AmazonInvalidInputException
  HResult=0x80131500
  Message=Report type 44900 does not support account ID of type class com.amazon.partner.account.id.VendorGroupId.
  Source=FikaAmazonAPI

Here's my code snippet:

ReportOptions ro = new ReportOptions();
            ro.Add("dateGranularity","WEEK");
            ro.Add("asinGranularity", "SKU");

            var parameters = new ParameterCreateReportSpecification();
            parameters.reportType = ReportTypes.GET_SALES_AND_TRAFFIC_REPORT;
            parameters.marketplaceIds = new MarketplaceIds();
            parameters.marketplaceIds.Add(MarketPlace.US.ID);
            parameters.dataStartTime = DateTime.Now.AddDays(-40);
            parameters.dataEndTime = DateTime.Now.AddDays(-30);
            parameters.reportOptions = ro;
            

            var reportId = await amazonConnection.Reports.CreateReportAsync(parameters);
            var filePath = string.Empty;
            string ReportDocumentId = string.Empty;

            while (string.IsNullOrEmpty(ReportDocumentId))
            {
                var reportData = await amazonConnection.Reports.GetReportAsync(reportId);
                if (!string.IsNullOrEmpty(reportData.ReportDocumentId))
                {
                    filePath = await amazonConnection.Reports.GetReportFileAsync(reportData.ReportDocumentId);
                    break;
                }
                if (reportData.ProcessingStatus == Report.ProcessingStatusEnum.FATAL)
                {
                    throw new Exception("Error with Generate report FATAL");
                }
                if (reportData.ProcessingStatus == Report.ProcessingStatusEnum.CANCELLED)
                {
                   // return null;
                }
                else Task.Delay(500).Wait();
            }

Thanks in advance for any ideas!

**Comment by wolendranh** [comment]
GET_SALES_AND_TRAFFIC_REPORT is seller report, seems like you are trying to create seller report using vendor_group_id which is vendor identifier.

**Reply by RestricePlanoax** [reply]
How did you check that we are using "vendor_group_id" ? 

I do not see it in the code, could you please point it to me. Thanks !

**Reply by wolendranh** [reply]
Sorry for such a late response. 

There is mention about it in error message you are getting:
```
Message=Report type 44900 does not support account ID of type class com.amazon.partner.account.id.VendorGroupId.
```

**Comment by RestricePlanoax** [comment]
I am still getting this error ! 

{
    "errors": [
        {
            "code": "InvalidInput",
            "message": "Report type 44900 does not support account ID of type class com.amazon.partner.account.id.VendorGroupId.",
            "details": ""
        }
    ]
}

can anyone help , please?

**Reply by MeenaAmz** [reply]
GET_SALES_AND_TRAFFIC_REPORT is seller report, you will not be able to retrieve a seller report with vendor credentials. Please refer to [reports doc](https://developer-docs.amazon.com/sp-api/docs/report-type-values#vendor-retail-analytics-reports) to understand the reports available for vendors.

**Comment by MeenaAmz** [comment, marked as answer]
GET_SALES_AND_TRAFFIC_REPORT is seller report, you will not be able to retrieve a seller report with vendor credentials. Please refer to [reports doc](https://developer-docs.amazon.com/sp-api/docs/report-type-values#vendor-retail-analytics-reports) to understand the reports available for vendors.
